nexus/pkg/gowandb/manager.go

```diff
@@ -15,22 +17,27 @@ type Manager struct {
 	addr string
 
 	// settings for all runs
-	settings *SettingsWrap
+	settings *settings.SettingsWrap
 }
 
 // NewManager creates a new manager with the given settings and responders.
-func NewManager(ctx context.Context, settings *SettingsWrap, addr string) *Manager {
+func NewManager(ctx context.Context, baseSettings *settings.SettingsWrap, addr string) *Manager {
 	manager := &Manager{
 		ctx:      ctx,
-		settings: settings,
+		settings: baseSettings,
 		addr:     addr,
 	}
 	return manager
 }
 
 func (m *Manager) NewRun() *Run {
 	conn := m.Connect(m.ctx)
-	run := NewRun(m.ctx, m.settings.Settings, conn)
+	// make a copy of the base manager settings
+	runSettings := m.settings.Copy()
+	if runSettings.RunId == nil {
+		runSettings.SetRunID(shared.ShortID(8))
+	}
```

Comment:
would be cool to move this into the settingswrap (like we do updates in python) <- not required for this PR